wlauto/workloads/iozone/__init__.py

```diff
@@ -0,0 +1,136 @@
+                                                             'iozone'))
+        Iozone.device_binary = self.device.install(Iozone.host_binary)
+
+    def setup(self, context):
+        self.results = os.path.join(self.device.working_directory,
+                                    iozone_results_txt)
+        self.command = self._build_command()
+
+        if self.threads and self.auto_mode:
+            raise ConfigError("You cannot set the number of threads and enable"
+                              " auto mode at the same time.")
+
+    def _build_command(self):
+        iozone_command = '{}'.format(self.device_binary)
+
+        if self.auto_mode:
+            self.auto_option = ' -a'
+            iozone_command += self.auto_option
+
+        if self.tests:
+            self.test_string = ''.join([' -i {}'.format(t) for t in self.tests])
+            iozone_command += self.test_string
+
+        if self.record_length > 0:
+            self.record_option = ' -r {}'.format(self.record_length)
+            iozone_command += self.record_option
+
+        if self.threads > 0:
+            self.threads_option = ' -t {}'.format(self.threads)
+            iozone_command += self.threads_option
+
+        if self.file_size > 0:
+            self.file_size_option = ' -s {}'.format(self.file_size)
+            iozone_command += self.file_size_option
+
+        if self.other_params:
+            other_params_string = ' ' + self.other_params
+            iozone_command += other_params_string
+
+        self.log_string = ' > {}'.format(self.results)
+        iozone_command += self.log_string
+
+        return iozone_command
+
+    def run(self, context):
+        self.device.execute(self.command, timeout=self.timeout)
+
+    def update_result(self, context):
+        self.device.pull_file(self.results, context.output_directory)
+        outfile = os.path.join(context.output_directory, iozone_results_txt)
+
+        with open(outfile, 'r') as iozone_file:
+            for line in iozone_file:
+                time_res_match = time_res_regex.search(line)
+                if time_res_match:
+                    context.result.add_metric("time_resolution",
+                                              float(time_res_match.group(1)),
+                                              time_res_match.group(2))
+
+                cache_size_match = cache_size_regex.search(line)
+                if cache_size_match:
+                    context.result.add_metric("processor_cache_size",
+                                              float(cache_size_match.group(1)),
+                                              cache_size_match.group(2))
+
+                cache_line_match = cache_line_regex.search(line)
+                if cache_line_match:
+                    context.result.add_metric("processor_cache_line_size",
+                                              float(cache_line_match.group(1)),
+                                              cache_line_match.group(2))
```

Comment:
I may be misunderstanding what iozone reports, but these seem to be the parameters under which the test was run, rather than the results of the test (so if anything, they should be added as classifiers), and the actual metrics aren't being extracted?